Nodejs/Product/Nodejs/Intellisense/ExpansionClient.cs

```diff
@@ -0,0 +1,310 @@
+                    baseIndentation = indentationLine.Substring(0, i);
+                    break;
+                }
+            }
+            return baseIndentation;
+        }
+
+        private void IndentSpan(ITextEdit edit, string indentation, int startLine, int endLine) {
+            var snapshot = _textView.TextBuffer.CurrentSnapshot;
+            for (int i = startLine; i <= endLine; i++) {
+                var curline = snapshot.GetLineFromLineNumber(i);
+                edit.Insert(curline.Start, indentation);
+            }
+        }
+
+        public int GetExpansionFunction(IXMLDOMNode xmlFunctionNode, string bstrFieldName, out IVsExpansionFunction pFunc) {
+            pFunc = null;
+            return VSConstants.S_OK;
+        }
+
+        public int IsValidKind(IVsTextLines pBuffer, TextSpan[] ts, string bstrKind, out int pfIsValidKind) {
+            pfIsValidKind = 1;
+            return VSConstants.S_OK;
+        }
+
+        public int IsValidType(IVsTextLines pBuffer, TextSpan[] ts, string[] rgTypes, int iCountTypes, out int pfIsValidType) {
+            pfIsValidType = 1;
+            return VSConstants.S_OK;
+        }
+
+        public int OnAfterInsertion(IVsExpansionSession pSession) {
+            return VSConstants.S_OK;
+        }
+
+        public int OnBeforeInsertion(IVsExpansionSession pSession) {
+            _session = pSession;
+            return VSConstants.S_OK;
+        }
+
+        public int OnItemChosen(string pszTitle, string pszPath) {
+            int caretLine, caretColumn;
+            GetCaretPosition(out caretLine, out caretColumn);
+
+            var textSpan = new TextSpan() { iStartLine = caretLine, iStartIndex = caretColumn, iEndLine = caretLine, iEndIndex = caretColumn };
+            return InsertNamedExpansion(pszTitle, pszPath, textSpan);
+
+        }
+
+        public int NextField() {
+            return _session.GoToNextExpansionField(0);
+        }
+
+        public int PreviousField() {
+            return _session.GoToPreviousExpansionField();
+        }
+
+        public int EndCurrentExpansion(bool leaveCaret) {
+            if (_selectEndSpan) {
+                TextSpan[] endSpan = new TextSpan[1];
+                if (ErrorHandler.Succeeded(_session.GetEndSpan(endSpan))) {
+                    var snapshot = _textView.TextBuffer.CurrentSnapshot;
+                    var startLine = snapshot.GetLineFromLineNumber(endSpan[0].iStartLine);
+                    var selectionLength = _selectionEnd.GetPosition(_textView.TextBuffer.CurrentSnapshot) - _selectionStart.GetPosition(_textView.TextBuffer.CurrentSnapshot);
+                    var span = new Span(startLine.Start + endSpan[0].iStartIndex + selectionLength, 0);
+                    _textView.Caret.MoveTo(new SnapshotPoint(snapshot, span.Start));
+                    _textView.Selection.Select(new SnapshotSpan(_textView.TextBuffer.CurrentSnapshot, span), false);

```

Comment:
since you're not actually selecting anything here, you should be able to simplify this to the ```_textView.Caret.MoveTo``` call with start being... ```startLine.Start + endSpan[0].iStartIndex + selectionLength```